src/LndClient/Data/Invoice.hs

```diff
@@ -33,6 +44,50 @@ instance FromGrpc Invoice GRPC.Invoice where
       <*> fromGrpc (GRPC.invoicePaymentRequest x)
       <*> fromGrpc (GRPC.invoicePrivate x)
       <*> fromGrpc (GRPC.invoiceAddIndex x)
-      <*> first
-        (\e -> FromGrpcError $ "Invalid Invoice State" <> show e)
-        (enumerated $ GRPC.invoiceState x)
+      <*> ( fromGrpc
+              =<< first
+                (\e -> FromGrpcError $ "Invalid Invoice State" <> show e)
+                (enumerated $ GRPC.invoiceState x)
+          )
+
+instance C2.FromGrpc Invoice LnGRPC.Invoice where
+  fromGrpc x =
+    Invoice
+      <$> invoiceRHash
+      <*> invoiceAmtPaidMsat
+      <*> invoiceValueMsat
+      <*> invoiceSettled
+      <*> invoiceSettleIndex
+      <*> invoiceMemo
+      <*> invoicePaymentRequest
+      <*> invoicePrivate
+      <*> invoiceAddIndex
+      <*> invoiceState
+    where
+      invoiceRHash = Right . RHash $ x ^. LnGRPC.rHash
```

Comment:
I think hardcoding parser implementation to every line is a bad practice. Just use `fromGrpc` or `C2.fromGrpc`. Such simple hardcoded implementations might miss some important features like byte size verification, or some other kind of verification.

Reply:
This was temporarily implementation, cleaned it up